jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1441,38 +1447,34 @@ ecma_builtin_typedarray_prototype_sort_compare_helper (ecma_value_t lhs, /**< le
 
   ecma_value_t compare_args[] = { lhs, rhs };
 
-  ECMA_TRY_CATCH (call_value,
-                  ecma_op_function_call (comparefn_obj_p,
-                                         ECMA_VALUE_UNDEFINED,
-                                         compare_args,
-                                         2),
-                  ret_value);
+  ecma_value_t call_value = ecma_op_function_call (comparefn_obj_p,
+                                                   ECMA_VALUE_UNDEFINED,
+                                                   compare_args,
+                                                   2);
 
-  if (!ecma_is_value_number (call_value))
+  if (ECMA_IS_VALUE_ERROR (call_value)
+      || ecma_is_value_number (call_value))
   {
-    ECMA_OP_TO_NUMBER_TRY_CATCH (ret_num, call_value, ret_value);
-    result = ret_num;
-    ECMA_OP_TO_NUMBER_FINALIZE (ret_num);
-
-    // If the coerced value can't be represented as a Number, compare them as equals.
-    if (ecma_number_is_nan (result))
-    {
-      result = ECMA_NUMBER_ZERO;
-    }
+    return call_value;
   }
-  else
+
+  ecma_number_t ret_num;
+
+  if (ECMA_IS_VALUE_ERROR (ecma_get_number (call_value, &ret_num)))
   {
-    result = ecma_get_number_from_value (call_value);
+    ecma_free_value (call_value);
+    return ECMA_VALUE_ERROR;
   }
```

Comment:
It would be good to reduce the number of "ecma_free_value` calls, maybe we can re-align this to something like this?:
```c
ecma_value_t number_result = ecma_get_number (call_value, &ret_num);
ecma_free_value (call_value);
if (ECMA_IS_VALUE_ERROR(number_result)) {
  return ECMA_VALUE_ERROR
}
```